Files/App.xaml.cs

```diff
@@ -394,6 +388,48 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
             deferral.Complete();
         }
 
+        // Occurs when an exception is not handled on the UI thread.
+        private static void OnUnhandledException(object sender, Windows.UI.Xaml.UnhandledExceptionEventArgs e)
+        {
+            Logger.Error(e.Exception, e.Message);
+
+            var appVersion = Package.Current.Id.Version;
+
+            var diagnosticInfo = new Dictionary<string, string>()
+            {
+                { "Message", e.Message },
+                { "Exception", e.Exception?.ToString() },
+                { "Culture", SystemInformation.Culture.EnglishName },
+                { "AvailableMemory", SystemInformation.AvailableMemory.ToString("F0") },
+                { "FirstUseTimeUTC", SystemInformation.FirstUseTime.ToUniversalTime().ToString("MM/dd/yyyy HH:mm:ss") },
+                { "OSArchitecture", SystemInformation.OperatingSystemArchitecture.ToString() },
+                { "OSVersion", SystemInformation.OperatingSystemVersion.ToString() },
+                { "AppVersion", string.Format("{0}.{1}.{2}.{3}", appVersion.Major, appVersion.Minor, appVersion.Build, appVersion.Revision) }
+            };
+
+            Analytics.TrackEvent("OnUnhandledException", diagnosticInfo);
```

Comment:
Instead of logging the crash as an event in AppCenter, perhaps it's better to save this in the local log file. AppCenter already catches the crash info and logging it as an event will make tracking our other analytics harder.

Reply:
Made changes.